### PR TITLE
Tidy sliding sync processing code

### PR DESCRIPTION
**Pull request by andybalaam**
### Clarify in tests that invite_state contains very minimal JSON

Supply JSON similar to what is found in MSC3575 for `invite_state`

### Simplify invited room sliding sync processing

Added some comments based on my understanding of MSC3575. Previously we
had a FIXME about how we were setting the room state to invite or join
based on the presence or absence of the invite_state property. I think
this behaviour is correct, so I added some comments explaining why.

The functional change here is to note that we call
store.get_or_create_stripped_room to find/create the stripped room, and
this actually deletes any room of this ID that is in the store, so our
later call to store.get_room would always fall back to getting the
stripped room, which we already have, so there was no need to do the
get_room call at all.

Applies on top of https://github.com/matrix-org/matrix-rust-sdk/pull/2039

### CODE REVIEW

File: crates/matrix-sdk-base/src/sliding_sync.rs
Comment (bnjbvr) [inline crates/matrix-sdk-base/src/sliding_sync.rs:217]: I can't see this code in `main`, and I just realized this is not based off `main` but another branch of yours, is this a mistake?

**Comment by bnjbvr** [comment]
Ohh I just saw that:

> Applies on top of https://github.com/matrix-org/matrix-rust-sdk/pull/2039

Alright, I'll wait for that one to be merged then :+1:

**Comment by andybalaam** [comment]
> Ohh I just saw that:
> 
> > Applies on top of #2039
> 
> Alright, I'll wait for that one to be merged then +1

I'll take the opportunity to force-push then, to tidy up after the merge of the even-earlier PR in the chain.

**Review by Hywan — APPROVED** [review]
Thanks for clarifying this part of the code. It looks good to me, and legit to consider the room as invited or joined based on the server's responses.

File: crates/matrix-sdk-base/src/sliding_sync.rs
Comment (andybalaam) [reply crates/matrix-sdk-base/src/sliding_sync.rs:217]: (As mentioned, it was part of a chain of PRs)

**Comment by andybalaam** [comment]
@Hywan all good to merge when the checks pass, thanks!

**Pull request merged**
